commit 5c3f1e46b61db591ea61d560ee51535286afa1a4
Author: Brandon Black <bblack@wikimedia.org>
Date:   Sat Nov 12 15:26:17 2016 +0000

    increase DEFAULT_BUFFER_SIZE from 4K to 8K
    
    The historical SSLeay value was 1K.  This was increased to 4K in
    for release 0.9.6b back in 2001.  Today, it is not uncommon for
    server handshake responses to exceed 4K given intermediate chain
    certs and OCSP stapling, but typical sizes are still within 8K.
    
    When the server response exceeds the buffer size, the TLS
    handshake suffers an extra RTT of latency penalty.  It is also
    difficult (and shouldn't be necessary given how common this is) to
    correct this manually using the normal APIs for a server
    application.
    
    There may have also been some rationale around TCP IW3 limits in
    setting the previous 4K size, but modern servers are typically
    IW10, which allows the server to initially send 8K easily.  IW10
    allows for at most 14600 in the ideal case, but I wouldn't expect
    it to commonly be any less than 12200 (IPv6 minimum MTU case).

diff -aur openssl11.orig/crypto/bio/bf_buff.c openssl11/crypto/bio/bf_buff.c
--- openssl11.orig/crypto/bio/bf_buff.c	2018-03-27 16:32:53.197232564 +0200
+++ openssl11/crypto/bio/bf_buff.c	2018-03-27 16:55:12.652712856 +0200
@@ -20,7 +20,7 @@
 static int buffer_new(BIO *h);
 static int buffer_free(BIO *data);
 static long buffer_callback_ctrl(BIO *h, int cmd, BIO_info_cb *fp);
-#define DEFAULT_BUFFER_SIZE     4096
+#define DEFAULT_BUFFER_SIZE     8192
 
 static const BIO_METHOD methods_buffer = {
     BIO_TYPE_BUFFER,
diff -aur openssl11.orig/doc/crypto/BIO_f_buffer.pod openssl11/doc/crypto/BIO_f_buffer.pod
--- openssl11.orig/doc/crypto/BIO_f_buffer.pod	2018-03-27 16:30:50.364425787 +0200
+++ openssl11/doc/crypto/BIO_f_buffer.pod	2018-03-27 16:55:34.672550858 +0200
@@ -37,7 +37,7 @@
 
 BIO_set_read_buffer_size(), BIO_set_write_buffer_size() and BIO_set_buffer_size()
 set the read, write or both read and write buffer sizes to B<size>. The initial
-buffer size is DEFAULT_BUFFER_SIZE, currently 4096. Any attempt to reduce the
+buffer size is DEFAULT_BUFFER_SIZE, currently 8192. Any attempt to reduce the
 buffer size below DEFAULT_BUFFER_SIZE is ignored. Any buffered data is cleared
 when the buffer is resized.
 
